controllers/storage/provision.go

```diff
@@ -0,0 +1,366 @@
+			"region":          []byte(m.HarborCluster.Spec.Storage.Oss.Region),
+			"bucket":          []byte(m.HarborCluster.Spec.Storage.Oss.Bucket),
+			"endpoint":        []byte(m.HarborCluster.Spec.Storage.Oss.Endpoint),
+			"internal":        []byte(m.HarborCluster.Spec.Storage.Oss.Internal),
+			"encrypt":         []byte(m.HarborCluster.Spec.Storage.Oss.Encrypt),
+			"secure":          []byte(m.HarborCluster.Spec.Storage.Oss.Secure),
+			"rootdirectory":   []byte(m.HarborCluster.Spec.Storage.Oss.Rootdirectory),
+			"chunksize":       []byte(m.HarborCluster.Spec.Storage.Oss.Chunksize)},
+	}
+}
+
+func (m *MinIOReconciler) ProvisionOss() (*lcm.Properties, error) {
+	OssSecret := m.generateOssSecret()
+	err := m.KubeClient.Create(m.Ctx, OssSecret)
+	p := &lcm.Property{
+		Name:  SwiftSecret,
+		Value: OssSecret.Name,
+	}
+	properties := &lcm.Properties{p}
+	return properties, err
+}
+
+func (m *MinIOReconciler) generateOssSecret() *corev1.Secret {
+	return &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Secret",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        m.HarborCluster.Name + "-" + DefaultExternalSecretSuffix,
+			Namespace:   m.HarborCluster.Namespace,
+			Labels:      m.getLabels(),
+			Annotations: m.generateAnnotations(),
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			"authurl":             []byte(m.HarborCluster.Spec.Storage.Swift.Authurl),
+			"username":            []byte(m.HarborCluster.Spec.Storage.Swift.Username),
+			"password":            []byte(m.HarborCluster.Spec.Storage.Swift.Password),
+			"container":           []byte(m.HarborCluster.Spec.Storage.Swift.Container),
+			"region":              []byte(m.HarborCluster.Spec.Storage.Swift.Region),
+			"tenant":              []byte(m.HarborCluster.Spec.Storage.Swift.Tenant),
+			"tenantid":            []byte(m.HarborCluster.Spec.Storage.Swift.Tenantid),
+			"domain":              []byte(m.HarborCluster.Spec.Storage.Swift.Domain),
+			"Domainid":            []byte(m.HarborCluster.Spec.Storage.Swift.Domainid),
+			"trustid":             []byte(m.HarborCluster.Spec.Storage.Swift.Trustid),
+			"insecureskipverify":  []byte(strconv.FormatBool(m.HarborCluster.Spec.Storage.Swift.Insecureskipverify)),
+			"prefix":              []byte(m.HarborCluster.Spec.Storage.Swift.Prefix),
+			"secretkey":           []byte(m.HarborCluster.Spec.Storage.Swift.Secretkey),
+			"authversion":         []byte(string(m.HarborCluster.Spec.Storage.Swift.Authversion)),
+			"endpointtype":        []byte(m.HarborCluster.Spec.Storage.Swift.Endpointtype),
+			"tempurlcontainerkey": []byte(strconv.FormatBool(m.HarborCluster.Spec.Storage.Swift.Tempurlcontainerkey)),
+			"tempurlmethods":      []byte(m.HarborCluster.Spec.Storage.Swift.Tempurlmethods),
+			"chunksize":           []byte(m.HarborCluster.Spec.Storage.Swift.Chunksize)},
+	}
+}
+
+func (m *MinIOReconciler) Provision() (*lcm.CRStatus, error) {
```

Comment:
How to use this method ?

Reply:
Provision inCluster are not implemented yet